workflows/diagnostics/fv3net/diagnostics/offline/views/plot.py

```diff
@@ -183,3 +185,66 @@ def plot_generic_data_array(
     plt.title(title)
     plt.tight_layout()
     return fig
+
+
+def plot_histogram(ds, varname: str, xscale="linear", yscale="linear"):
+    """Plot 1D histogram of varname."""
+
+    fig, ax = plt.subplots()
+    bin_name = varname.replace("histogram", "bins")
+
+    units = units_from_name(varname.replace("_histogram", ""))
+
+    for source in ["target", "predict"]:
+        v = ds[varname].sel({DERIVATION_DIM_NAME: source})
+        ax.step(v[bin_name], v, where="post", linewidth=1, label=source)
+    ax.set_xlabel(f"{getattr(v, 'long_name', v.name)} {units}")
+    ax.set_ylabel(f"Frequency {units}^-1")
+    ax.set_xscale(xscale)
+    ax.set_yscale(yscale)
+    ax.set_ylim([0, None])
+    ax.legend()
+    ax.set_title(source)
+    fig.tight_layout()
+    return fig
+
+
+def plot_histogram2d(ds, xname: str, yname: str):
+    """Plot 2D histogram of xname versus yname."""
+
+    count_name = f"{xname.lower()}_versus_{yname.lower()}_hist_2d"
+    conditional_average_name = f"conditional_average_of_{yname}_on_{xname}"
+    x_bin_name = f"{xname}_bins"
+    y_bin_name = f"{yname}_bins"
+    x_bin_widths_name = f"{xname.lower()}_bin_width_hist_2d"
+    y_bin_widths_name = f"{yname.lower()}_bin_width_hist_2d"
+    fig = plt.figure(figsize=(8, 4))
```

Comment:
the current figure size makes it a bit hard to read.

Reply:
Fixed